aip/general/0191.md

```diff
@@ -127,8 +127,21 @@ which don't include compound names, be aware that _adding_ this annotation
 (with a value not equivalent to the default) constitutes a breaking change in
 that language. When releasing protos, be sure that omissions are intentional.
 
+## Rationale
+
+### Java packaging options
+
+Setting the option `java_multiple_files` to `true` is the convention, because 
+creates a cleaner class file structure (one Protobuf type per class file), and
+more precise importants. As a result, `java_outer_classname` is  necessary in
+order to prevent a potential naming collision between the class generated for
+the containing `.proto` file (the "outer" class) and any potential class files
+generated for a type therein e.g. `foo.proto` containing `message Foo` and both
```

Comment:
Firstly, I would remove the "e.g." and, in a new sentence, use "For example,".

Secondly, I don't think your example is clear or precise. It is perfectly natural, in fact expected, to have the file "Foo.java" contain `class Foo`. I say that it is expected because ever Java file has to have at least one outer class named after the file that has visibility that is package-private or public. Maybe see if you can think of a better example?

Reply:
The point is that there are two different files named `Foo.java` both containing `class Foo` b.c the `foo.proto` file gets one class file and the message therein `message Foo` gets it's own class file --> collision.

Reply:
I've changed to Book as its more common, and I've tried to be more explicit in where the collision happens. PTAL